File: my-life-backend/services/user-service/src/main/java/com/mylife/user/service/AuthService.java
package com.mylife.user.service;

import com.mylife.user.dto.request.LoginRequest;
import com.mylife.user.dto.request.RegisterRequest;
import com.mylife.user.dto.response.AuthResponse;

public interface AuthService {
    void register(RegisterRequest request);

    AuthResponse login(LoginRequest request);
    AuthResponse createRefreshToken(String refreshToken);
}

File: my-life-app/services/user/AuthService.ts
// src/services/AuthService.ts

import { http } from "@/http/http_helper";
import { ApiResponse } from "@/types/api_response";
import { LoginRequest } from "@/types/dto/user/request/login";
import { RegisterRequest } from "@/types/dto/user/request/register";
import { AuthResponse } from "@/types/dto/user/response/auth_response";
const BASE = "/api/proxy/user-service/auth";
export class AuthService {
    static login(data: LoginRequest) {
        return http.post<ApiResponse<AuthResponse>>(`${BASE}/login`, data);
    }

    static register(data: RegisterRequest) {
        return http.post<ApiResponse<string>>(`${BASE}/register`, data);
    }
}
